File: web/domain/filing/models/deductions/medical/deduction-80ddb-model.ts
/**
 * Section 80DDB Deduction Interface (Medical Treatment)
 * Mirrors: section_80ddb in FilingModel
 */

export interface Deduction80DDBModel {
  deductionId?: number | null
  filingId: number | null
  treatmentFor: string  // "Self" / "Dependant"
  seniorCitizenType?: string | null  // "Self" / "Dependant"
  disease?: string | null
  expenditureIncurred: number
}


export const INITIAL_80DDB_FORM_DATA: Deduction80DDBModel = {
  deductionId: null,
  filingId: null,
  treatmentFor: 'Self',
  seniorCitizenType: 'Self  ',
  disease: '',
  expenditureIncurred: 0,
}

File: api/filing/models/deduction_80ddb_model.py
from .api_base_model import ApiBaseModel


class Deduction80DDBModel(ApiBaseModel):
    deduction_id: int | None = None
    filing_id: int
    treatment_for: str | None = None  # "Self" or "Dependant"
    senior_citizen_type: str | None = None  # "Self" or "Dependant"
    disease: str | None = None
    expenditure_incurred: float
# Deduction80DDBModel